fastapi_backend/tests/conftest.py

```diff
@@ -37,14 +39,62 @@ async def db_session():
 async def test_client(db_session):
     """Fixture to create a test client that uses the test database session."""
 
+    # Override for user database
     async def override_get_db():
         session = SQLAlchemyUserDatabase(db_session, User)
         try:
             yield session
         finally:
             await db_session.close()
 
+    # Override for general database session
+    async def override_get_async_session():
+        try:
+            yield db_session
+        finally:
+            await db_session.close()
```

Comment:
Why do we need this override? Please clarify on the comment.

Reply:
1. We need the override in general to use the test_database instead of the regular database.

We have these two overrides due to fastapiusers that uses the get_user_db as a dependecy. Fastapi_users uses that into their endpints. It's a wrapper that adds more methods to the regular session.

If we weren't using fastapi_users, we would need a single override.